src/components/Form/Form.js

```diff
@@ -0,0 +1,78 @@
+import React from 'react'
+import PropTypes from 'prop-types'
+
+import { useFormState } from '../../hooks/useFormState'
+
+export function Form({ children, config, render }) {
+  const inputNames = Object.keys(config.inputs)
+
+  // Set up initial values
+  let initialValues = {}
+  inputNames.forEach((x) => {
+    initialValues[x] = 'err' // by default inputs have "hidden" errors
+  })
+
+  // Hooks
+  const [
+    getInputErrors,
+    getInputValues,
+    setStateFactory,
+    getFormErrorMessage,
+    setFormErrorMessage,
+    getFormIsValid,
+  ] = useFormState(initialValues)
+
+  // Validates form, sets button to disabled if there's an error
+  function inputNamePropsTransformer(inputName) {
+    return {
+      name: inputName,
+      formChangeHandler: setStateFactory(inputName),
+      validator: (input) =>
+        config.inputs[inputName].validators
+          .reduce((errors, validator) => errors.concat(validator(input)), [])
+          .filter((x) => !!x) // remove empty strings
+          .join(', '),
+      labelCopy: config.inputs[inputName].labelCopy,
+      ['data-tid']: [config.formName, config.formId, inputName].join('-'),
+    }
+  }
+
+  async function onSubmit(syntheticReactEvent) {
+    setFormErrorMessage('')
+
+    syntheticReactEvent.preventDefault()
+
+    if (!getFormIsValid()) {
+      // Should already be handled by form itself, but...
+      if (getInputErrors()) {
```

Comment:
Can you add an example of `getInputErrors` or put some 
```js
//example code I can plop into styleguidist live editor here??
```

Reply:
sure one sec

Reply:
well i'll do it in form-ii

Reply:
did it over there, basically the submit isn't set to display as `disabled={true}`. In this example form its disabled as long as the form is untouched and has errors.